crates/ruff_python_codegen/src/generator.rs

```diff
@@ -1726,6 +1741,62 @@ class Foo:
         assert_eq!(round_trip(r#""he\"llo""#), r#"'he"llo'"#);
```

Comment:
I nearly suggested a silly change to `BytesRepr` and `StrRepr` that would have broken some edge cases. Luckily some tests failed -- but only for the `StrRepr` change; no tests failed for the `BytesRepr` change. Here's a test that would have failed on my bad patch, which might be good to add:

```suggestion
        assert_eq!(round_trip(r#""he\"llo""#), r#"'he"llo'"#);
        assert_eq!(round_trip(r#"b"he\"llo""#), r#"b'he"llo'"#);
```